tests/sentry/event_manager/test_event_manager.py

```diff
@@ -1618,6 +1618,35 @@ def test_category_match_group(self):
 
         assert event1.get_hashes().hashes == event2.get_hashes(grouping_config).hashes
 
+    def test_synthetic_exception_detection(self):
```

Comment:
fwiw ideally this test wouldn't access the database. I wish _calculate_event_grouping would be unit-testable

Reply:
...then we wouldn't have to keep updating the test harness in test_variants too